src/lib/y2storage/fake_device_factory.rb

```diff
@@ -609,13 +609,15 @@ def create_lvm_pv(parent, args)
     def create_btrfs(parent, args)
       log.info("#{__method__}( #{parent}, #{args} )")
       default_subvolume = args["default_subvolume"]
-      if default_subvolume
-        blk_device = Y2Storage::BlkDevice.find_by_name(@devicegraph, parent)
+      if default_subvolume && !default_subvolume.empty?
+        blk_device = BlkDevice.find_by_name(devicegraph, parent)
         filesystem = blk_device.filesystem
         if !filesystem || !filesystem.type.is?(:btrfs)
           raise HierarchyError, "No btrfs on #{parent}"
         end
-        filesystem.create_btrfs_subvolume(@default_subvolume)
+        btrfs = Storage.to_btrfs(filesystem.to_storage_value)
```

Comment:
That was the fix I did at 20:30 to bring this whole thing back to life after rebasing when it started to segfault or throw bad_cast exceptions. Can you understand I lost patience after spending **the whole day** trying to use the YAML scenarios for that thing?